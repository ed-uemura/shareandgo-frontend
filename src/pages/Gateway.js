import React from 'react';
import ShareGoLogo from '../images/ShareGoLogo.svg';
import ButtonPrimary from '../components/ButtonPrimary';
import ButtonSecondary from '../components/ButtonSecondary';

const Gateway = () => {
	// check if user is logged in
	const localStorage = window.localStorage;
	const userID = localStorage.getItem("userID");
	const userType = localStorage.getItem("userType");
	if(userID && userType) {
		// redirect to each user type's main page
		if(userType === "passenger") {
			window.location.href = "/main-passenger";
		} else if(userType === "driver") {
			window.location.href = "/main-driver";
		}
	}
	else{
		// cleanup session
		localStorage.setItem("userToken", null);
		localStorage.setItem("userID", null);
		localStorage.setItem("userFirstName", null);
		localStorage.setItem("userType", null);
	}
	return (
		<>
			<section className='col-10 col-sm-6 col-md-6 col-lg-4 position-absolute top-50 start-50 translate-middle'>
				<figure className='text-center'>
					<img
						src={ShareGoLogo}
						className='mb-3 img-fluid'
						alt='Share & Go Logo'
					/>
				</figure>
				<div className='row row-cols-2 g-3 text-center'>
					<div className='col'>
						<ButtonPrimary
							text='Register'
							className='col-12'
							link='/registration'
							clickAction=''
						/>
					</div>
					<div className='col'>
						<ButtonSecondary
							text='Login'
							className='col-12'
							link='/login'
							clickAction=''
						/>
					</div>
				</div>
			</section>
		</>
	);
};

export default Gateway;
