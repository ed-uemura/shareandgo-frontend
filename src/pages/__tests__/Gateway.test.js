import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import Gateway from '../Gateway';
import { MemoryRouter } from 'react-router-dom';

afterEach(() =>{
    cleanup();
});

test('login button should be rendered', () =>{
    render(<MemoryRouter>
        <Gateway />
    </MemoryRouter>)
    const loginButtonEl = screen.getByText(/login/i);
    expect(loginButtonEl).toBeInTheDocument();
})

test('register button should be rendered', () =>{
    render(<MemoryRouter> 
        <Gateway/>
        </MemoryRouter>);
    const registerButtonEl = screen.getByText(/register/i)
    expect(registerButtonEl).toBeInTheDocument();
})

test('logo should be rendered', async() =>{
    const {getByAltText} = await render(  
        <MemoryRouter> 
        <Gateway/>
        </MemoryRouter>     
        );
    const image = getByAltText('Share & Go Logo');
    expect(image).toBeInTheDocument();
});

test('matches snapshot', () =>{ 
    const tree = renderer.create(
        <MemoryRouter>
            <Gateway />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});