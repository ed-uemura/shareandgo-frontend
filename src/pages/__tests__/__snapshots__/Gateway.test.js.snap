// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="col-10 col-sm-6 col-md-6 col-lg-4 position-absolute top-50 start-50 translate-middle"
>
  <figure
    className="text-center"
  >
    <img
      alt="Share & Go Logo"
      className="mb-3 img-fluid"
      src="ShareGoLogo.svg"
    />
  </figure>
  <div
    className="row row-cols-2 g-3 text-center"
  >
    <div
      className="col"
    >
      <a
        className="btn btn-primary col-12"
        href="/registration"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
    <div
      className="col"
    >
      <a
        className="btn btn-secondary col-12"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </div>
</section>
`;
